cpp/src/reshape/interleave_columns.cu

```diff
@@ -19,16 +19,102 @@
 #include <cudf/detail/nvtx/ranges.hpp>
 #include <cudf/table/table_device_view.cuh>
 #include <cudf/types.hpp>
+#include <strings/utilities.cuh>
 
 namespace cudf {
 namespace detail {
 namespace {
 struct interleave_columns_functor {
   template <typename T, typename... Args>
-  std::enable_if_t<not cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>> operator()(
-    Args&&... args)
+  std::enable_if_t<not cudf::is_fixed_width<T>() and not std::is_same<T, cudf::string_view>::value,
+                   std::unique_ptr<cudf::column>>
+  operator()(Args&&... args)
   {
-    CUDF_FAIL("Only fixed-width types are supported in interleave_columns.");
+    CUDF_FAIL("interleave_columns not supported for dictionary and list types.");
+  }
+
+  template <typename T>
+  std::enable_if_t<std::is_same<T, cudf::string_view>::value, std::unique_ptr<cudf::column>>
+  operator()(table_view const& strings_columns,
+             bool create_mask,
+             rmm::mr::device_memory_resource* mr,
+             cudaStream_t stream = 0)
```

Comment:
@harrism This stream argument is only passable via the `type_dispatcher`. Do we always want to add detail APIs that match the public facing API (but also have allow for passing a string parameter)?

Reply:
the common convention i see elsewhere (for the most part) is for the detail api to have the `stream` parameter defaulted to `0` in the `detail` api, although this is flouted in some places.

Reply:
In general yes we need to plumb streams into the depths of the API. Typically the MR needs to go last though, and we are planning to eliminate the default 0 stream value, but that is under discussion and in progress.